Sources/ReplyService/FacebookGraph.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  FacebookGraph.swift
+//  SwiftBot
+//
+//  Created by Andrew Tokarev on 27/06/2017.
+//
+//
+
+import Foundation
+import PerfectCURL
+
+enum FacebookGraphRequestHttpMethod: Hashable, CustomStringConvertible {
+    
+    case get
+    case post
+    
+    public var description: String {
+        switch self {
+        case .get:      return "GET"
+        case .post:     return "POST"
+        }
+    }
+}
+
+struct FacebookGraphRequest {
+    
+    let path: String                // /me/messages
+    let params: [String: String]    // access_token:xxx
+    let body: Data                  // might be json
+    let contentType: String         // application/json
+    
+    let httpMethod: FacebookGraphRequestHttpMethod
+    
+    static func meMessagePostRequest(accessToken: String, jsonBody: Data) -> FacebookGraphRequest {
+        return FacebookGraphRequest(path: "/me/messages",
+                                    params: ["access_token": accessToken],
+                                    body: jsonBody,
+                                    contentType: "application/json",
+                                    httpMethod: .post)
+    }
+}
+
+struct FacebookGraphResponse {
+    public let body : String?
+    public let error: Error?
+    
+    init(body: String) {
+        self.body  = body
+        self.error = nil;
+    }
+    
+    init(error: Error) {
+        self.body  = nil;
+        self.error = error
+    }
+}
+
+final class FacebookGraph {
```

Comment:
can be done, 
Initially I implemented it like that, but wanted to avoid possible naming collision with full fledged library.